core/src/main/scala/quasar/destination/gbq/GBQDestination.scala

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+import slamdata.Predef._
+
+import quasar.api.destination.{Destination, DestinationError, DestinationType, ResultSink}
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.connector.{MonadResourceErr, ResourceError}
+import quasar.api.push.RenderConfig
+import quasar.api.resource.ResourceName
+import quasar.api.table.{ColumnType, TableColumn}
+
+import argonaut._, Argonaut._
+
+import cats.ApplicativeError
+import cats.effect.{Concurrent, ContextShift, Resource, Sync}
+import cats.implicits._
+
+import com.google.auth.oauth2.AccessToken
+
+import eu.timepit.refined.auto._
+
+import fs2.Stream
+
+import org.http4s.{
+  AuthScheme,
+  Credentials,
+  EntityEncoder,
+  MediaType,
+  Method,
+  Request,
+  Status,
+  Uri}
+import org.http4s.argonaut.jsonEncoderOf
+import org.http4s.client.Client
+import org.http4s.headers.{Authorization, `Content-Type`, Location}
+
+import org.slf4s.Logging
+
+import scalaz.NonEmptyList
+
+import shims._
+
+final class GBQDestination[F[_]: Concurrent: ContextShift: MonadResourceErr](
+    client: Client[F], 
+    config: GBQConfig,
+    sanitizedConfig: Json) extends Destination[F] with Logging {
+
+  def destinationType: DestinationType = DestinationType("gbq", 1L)
+
+  def sinks: NonEmptyList[ResultSink[F]] = NonEmptyList(gbqSink)
+
+  private def gbqSink: ResultSink[F] = 
+    ResultSink.csv[F](RenderConfig.Csv()) { (path, columns, bytes) =>
+      val gbqSchema = tblColumnToGBQSchema(columns)
+      
+      val tableNameF = path.uncons match {
+        case Some((ResourceName(name), _)) => name.pure[F]
+        case _ => MonadResourceErr[F].raiseError[String](
+          ResourceError.notAResource(path))
+      }
+
+      for {
+        tableName <- Stream.eval[F, String](tableNameF)
+        gbqJobConfig = formGBQJobConfig(gbqSchema, config, tableName)
+        accessToken <- Stream.eval(GBQAccessToken.token(config.authCfg.getBytes("UTF-8")))
+        _ <- Stream.eval((mkDataset(client, accessToken, config)))
+        eitherloc <- Stream.eval(mkGbqJob(client, accessToken, gbqJobConfig))
+        _ <- Stream.eval(
+          Sync[F].delay(
+            log.info(s"(re)creating ${config.project}.${config.datasetId}.${tableName} with schema ${columns.show}")))
+        _ <- eitherloc match {
+            case Right(locationUri) => {
```

Comment:
Style note: you don't need curly braces here

Reply:
minor style note